### DISCUSSION TITLE
Passcode options not available

### DISCUSSION THREAD

**Discussion by iosmoon38** [orchetect/swift-osc#30] [Q&A]
let oscMessage = OSCMessage(
            "/test/play",
            values: ["Test string", 123]
        )
        do {
            try oscClient.send(
                oscMessage,
                to: "192.168.1.234", // IP address or hostname
                port: 60088 // standard OSC port but can be changed
            )
        }catch {
            print(error.localizedDescription)
        }
        
        
   **There are no options for using a passcode also some ports are not working for send commands e.g. 60088.**
   And how we can send message type TCP or UDP?

Please guide me if you know the same.

Thanks in advance.

**Comment by orchetect** [comment]
I'm not sure what passcode is referring to. There is no provision specifically for passwords or encryption in the OSC specification.

OSC is primarily a plain-text encoding that is sent over UDP. OSCKit provides a UDP `OSCClient` and `OSCServer`. It's possible to transmit over TCP but that would be a custom implementation.
